flink-runtime/src/main/java/org/apache/flink/runtime/metrics/MetricRegistryImpl.java

```diff
@@ -463,7 +463,7 @@ public void addSpan(SpanBuilder spanBuilder) {
             if (LOG.isTraceEnabled()) {
                 LOG.trace("addSpan");
             }
-            if (reporters != null) {
+            if (traceReporters != null) {
```

Comment:
Looks good - but can we unit test this change?

Reply:
Thank you for comment!
I also checked `MetricRegistryImplTest` before push this commit, but there was no tests for null cases.
As mentioned in pr summary, I thought this can't be reproducible because all reporters values are initialized to empty `ArrayList`.
If you let me know a good way to reproduce cases, I will apply that.

However, this change is no longer valid because https://github.com/apache/flink/pull/26316, which is expected to in deployed in version 2.1, was merged.
If you don't mind for applying this change, could I change the base branch to `release-2.0`?